### PR TITLE
Bypass 16kb userdata limit by uploading to s3 bucket

### PR DESCRIPTION
**Pull request by redbaron**
Crude fix for #252 .

This is a best-effort rebase+squash of our internal branch on top of master.  It seems to solve this unexpected blocker for us, but code is brittle and moderately ugly, but not much more that already existing code around there :) I have no capacity to refine it further. It _breaks_ nodepools (another reason to get rid of them!).

Probably best thing to do is to make `cluster.yml` to accept `s3Bucket` config option and remove `s3-uri` flag, making presence of a readable bucket for cluster needs a hard requirement.

As I said, I wont be able to make it better, please adopt it, edit it to your needs and fix it,throw away it or whatever, but fix 16KB problem in master one way or another It is a major blocker for us so had to be solved ASAP, I imagine other users feel this pain too.

No testing of this PR is done whatsoever, just hacked it so that it compiles. Internal branch this PR is based on seems to be working fine.

### CODE REVIEW

**Comment by redbaron** [comment]
small comment on PR itself: all it is doing is plumbing `stackConfig` (which includes what of *Cluster flavours, I lost track which one)all the way up to a point where CF template upload is happening.

File: config/config.go [hunk 1]
```diff
@@ -577,7 +577,7 @@ func (c Cluster) stackConfig(opts StackTemplateOptions, compressUserData bool) (
 		stackConfig.Config.TLSConfig = compactAssets
 	}
 
-	if stackConfig.UserDataWorker, err = userdatatemplate.GetString(opts.WorkerTmplFile, stackConfig.Config, compressUserData); err != nil {
+	if stackConfig.UserDataWorker, err = userdatatemplate.GetString(opts.WorkerTmplFile, stackConfig.Config, false); err != nil {
```
Comment (mumoshu) [inline config/config.go]: Thanks! I'd been bitten by this myself 😛

**Comment by mumoshu** [comment]
Thanks for the follow-ups!
I'm testing/refactoring this in my internal branch.
I'll merge this and/or my branch including your commit probably after #227 and/or #169

**Comment by mumoshu** [comment]
Closing this in favor of #268(It is almost ready to be merged!

**Pull request closed without merging**